pytest-embedded-idf/pytest_embedded_idf/idf_unity_tester.py

```diff
@@ -1,13 +1,40 @@
-# SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
-# SPDX-License-Identifier: Apache-2.0
+import logging
 import time
+from functools import wraps
 from threading import Semaphore, Thread
 from typing import List, Union
 
 from pexpect.exceptions import TIMEOUT
 from pytest_embedded import Dut, unity, utils
 from pytest_embedded_idf.dut import UnittestMenuCase
 
+DEFAULT_RETRY_COUNT = 3
+DEFAULT_TIMEOUT = 90
+
+
+def timeit(func):
+    @wraps(func)
+    def timeit_wrapper(*args, **kwargs):
+        retry_count = int(kwargs.get('timeout', DEFAULT_RETRY_COUNT))
```

Comment:
- The `retry_count` here should be defined at `UnittestMenuCase.attributes['retry']`
- timeout should be defined at `UnittestMenuCase.attributes['timeout']`

> the original idea of the retry is for those flaky unit test cases. it should be parsed from the test case's full name. like TEST_CASE(...[retry=3][timeout=120])

As I mentioned above, these keywords should be defined in the `TEST_CASE(...)` c file. Not a keyword for python function.